src/main/java/com/microsoft/azure/vmagent/util/TokenCache.java

```diff
@@ -103,16 +93,23 @@ public static ApplicationTokenCredentials get(final AzureCredentials.ServicePrin
     }
 
     public Azure getAzureClient() throws AzureCloudException {
-        try {
-            return Azure
-                    .configure()
-                    .withInterceptor(new AzureVMAgentPlugin.AzureTelemetryInterceptor())
-                    .withLogLevel(Constants.DEFAULT_AZURE_SDK_LOGGING_LEVEL)
-                    .withUserAgent(getUserAgent())
-                    .authenticate(get(credentials))
-                    .withSubscription(credentials.getSubscriptionId());
-        } catch (Exception e) {
-            throw AzureCloudException.create(e);
+        if (client == null) {
+            try {
+                synchronized (TSAFE) {
+                    if (client == null) {
+                        client = Azure
+                                .configure()
+                                .withInterceptor(new AzureVMAgentPlugin.AzureTelemetryInterceptor())
+                                .withLogLevel(Constants.DEFAULT_AZURE_SDK_LOGGING_LEVEL)
+                                .withUserAgent(getUserAgent())
+                                .authenticate(get(credentials))
```

Comment:
What does the `authenticate` method actually perform? I think it's better to go deeper. If `authenticate` talks with azure to get/refresh something like `AccessToken` which will expire in certain time, it could be a problem with the `client`.